### PR TITLE
feat: [#176205711] Allow the script to run multiple times

### PR DESCRIPTION
**Pull request by gquadrati**
Minor changes to let the user run the script multiple times for updates

### CODE REVIEW

**Comment by balanza** [comment]
I've added a small fix in https://github.com/pagopa/developer-laptop/pull/2/commits/808fbbb3536ff4c78d2a93e78eff56cb1367da7c, please check it.

Besides that, I've run the script on my laptop and it runs smoothly. I noticed the following:
1. `vscode` settings are overwrite by the defaults. That means that I've lost all the changes I've done from the previous update. As `vscode` settings are very personal, we should reason about merging the default setting with what we have already
1. The very last output I receive in the terminal is _`Installing omyzsh...`_, then I have no more feedback. I didn't understand if the script blocked or it just piped the output somewhere else

Point 2 should be investigated, to be sure we're not damaging the existing installation. About point 1, we can either solve it now or do it in another PR

**Comment by gquadrati** [comment]
> I've added a small fix in [808fbbb](https://github.com/pagopa/developer-laptop/commit/808fbbb3536ff4c78d2a93e78eff56cb1367da7c), please check it.
> 
> Besides that, I've run the script on my laptop and it runs smoothly. I noticed the following:
> 
> 1. `vscode` settings are overwrite by the defaults. That means that I've lost all the changes I've done from the previous update. As `vscode` settings are very personal, we should reason about merging the default setting with what we have already
> 2. The very last output I receive in the terminal is _`Installing omyzsh...`_, then I have no more feedback. I didn't understand if the script blocked or it just piped the output somewhere else
> 
> Point 2 should be investigated, to be sure we're not damaging the existing installation. About point 1, we can either solve it now or do it in another PR

I'll fix the 2nd, and move point 1 to another PR.

**Comment by gquadrati** [comment]
Please, try it now.

File: mac [hunk 1]
```diff
@@ -237,9 +237,10 @@ fancy_echo "Visual Studio Code configured."
 fancy_echo "Configuring nodenv..."
 eval "nodenv init -"
 
-# nodenv patch
+# add nodenv initialization to the shell file, if not already present
 fancy_echo "patching nodenv init into .zshrc file"
-echo "eval \"\$(nodenv init -)\"" >> .zshrc
+nodenv_init="eval \"\$(nodenv init -)\""
+cat "$HOME/.zshrc" | grep "$nodenv_init" > /dev/null || echo "$nodenv_init" >> "$HOME/.zshrc"
```
Comment (balanza) [inline mac:254]: we append the init script only if there's no already

File: mac [hunk 2]
```diff
@@ -279,13 +280,15 @@ fi
...
 
 if [ ! -d "$SPACESHIP_ROOT" ]; then
-  git clone https://github.com/denysdovhan/spaceship-prompt.git "$SPACESHIP_ROOT"
+  fancy_echo "Installing spaceship..."
+  git clone https://github.com/denysdovhan/spaceship-prompt.git /Users/emanuele/.oh-my-zsh/custom/themes/spaceship-prompt
   ln -s "$SPACESHIP_ROOT/spaceship.zsh-theme" "$ZSH_CUSTOM/themes/spaceship.zsh-theme"
+  fancy_echo "Setting spaceship as default theme..."
   sed -i "" "s/^ZSH_THEME=.*/ZSH_THEME=\"spaceship\"/g" "$HOME/.zshrc"
 else
+  fancy_echo "Updating spaceship..."
```
Comment (balanza) [inline mac]: I just moved messages

File: mac (hunk 1 above)
Comment (gquadrati) [reply mac:254]: you're right, much better.. and also using $HOME allows us to run the script wherever we want!

**Comment by balanza** [comment]
LGTM

**Pull request merged**